src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/AS_StrandBiasTest.java

```diff
@@ -0,0 +1,229 @@
+package org.broadinstitute.hellbender.tools.walkers.annotator;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.GenotypesContext;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFInfoHeaderLine;
+import org.apache.log4j.Logger;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.utils.genotyper.MostLikelyAllele;
+import org.broadinstitute.hellbender.utils.genotyper.PerReadAlleleLikelihoodMap;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFHeaderLines;
+
+import java.util.*;
+
+/**
+ * Allele-specific implementation of strand bias annotations
+ */
+public abstract class AS_StrandBiasTest extends StrandBiasTest implements ReducibleAnnotation {
+    private final static Logger logger = Logger.getLogger(AS_StrandBiasTest.class);
+    protected static final String splitDelim = "\\|"; //String.split takes a regex, so we need to escape the pipe
+    protected static final String printDelim = "|";
+    protected static final String reducedDelim = ",";
+    protected static final int MIN_COUNT = 2;
+    protected static final double MIN_PVALUE = 1E-320;
+    protected static final int FORWARD = 0;
+    protected static final int REVERSE = 1;
+    protected final List<Integer> ZERO_LIST = new ArrayList<>();
+
+    public AS_StrandBiasTest(){
+        ZERO_LIST.add(0,0);
+        ZERO_LIST.add(1,0);
+    }
+
+    @Override
+    public List<VCFInfoHeaderLine> getDescriptions() {
+        //TODO only raw for now
+//        if (AnnotationUtils.walkerRequiresRawData(callingWalker))
+            return Arrays.asList(GATKVCFHeaderLines.getInfoLine(getRawKeyName()));
+//        else
+//            return Arrays.asList(GATKVCFHeaderLines.getInfoLine(getKeyNames().get(0)));
+    }
+
+    @Override
+    public String getRawKeyName() { return GATKVCFConstants.AS_SB_TABLE_KEY; }
+
+    @Override
+    public Map<String, Object> annotate(final ReferenceContext ref,
+                                        final VariantContext vc,
+                                        final Map<String, PerReadAlleleLikelihoodMap> stratifiedPerReadAlleleLikelihoodMap) {
+        return annotateRawData(ref, vc, stratifiedPerReadAlleleLikelihoodMap);
+    }
+
+    @Override
+    public Map<String, Object> annotateRawData(final ReferenceContext ref,
+                                               final VariantContext vc,
+                                               final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap ) {
+
+        //for allele-specific annotations we only call from HC and we only use perReadAlleleLikelihoodMap
+        if ( perReadAlleleLikelihoodMap == null) {
+            return new HashMap<>();
```

Comment:
You are inconsistent on the return value in this case -- some annotations return an empty `Map`, others return null. Stick to one or the other and document behavior in method contract.